### DISCUSSION TITLE
MPX5010DP датчик давления

### DISCUSSION THREAD

**Discussion by alcoroll** [LKosoj/Samovar#7] [Q&A] [answered]
приветствую, у меня есть датчик давления MPX5010DP , он более удобен для использования в аппарате , т.к. проще монтируется в крышку бака (из-за своей формы) 
как я могу его использовать с данным скетчем ? какие могут быть технические ограничения? 

![MPX5010DP-Pinout](https://user-images.githubusercontent.com/44224706/217276657-5ab78c6f-e709-4999-8a96-46039cbc2d09.jpg)

**Comment by LKosoj** [comment]
Технических ограничений нет. Можно подключить датчик к 34 пину, он не задействован на текущий момент и читать аналоговый сигнал, возвращаемый датчиком.
Дальше есть два варианта:
1. Доработать скетч, чтобы в интерфейсе показывалась разность давлений и реализовать алгоритм, по которому Самовар должен что-то делать в случае изменения давления.
2. Использовать Lua-скрипт, в котором считывать показания датчика и в зависимости от этих показаний управлять Самоваром. Показания так же можно выводить в интерфейс. Для прототипирования это самый простой вариант, так как алгоритм работы можно менять в процессе, не меняя скетч.

**Comment by LKosoj** [comment, marked as answer]
Добавлена поддержка датчика давления
